mailscanner/functions.php

```diff
@@ -352,172 +397,173 @@ function html_start($title, $refresh = 0, $cacheable = true, $report = false)
+                $msg = new xmlrpcmsg('postfix_queues', array());
+                $rsp = xmlrpc_wrapper($servers[$i], $msg);
+                if ($rsp->faultCode() === 0) {
+                    $response = php_xmlrpc_decode($rsp->value());
+                    $inq += $response['inq'];
+                    $outq += $response['outq'];
                 } else {
-                    // Not activated because this need to be tested.
-                    //$cmd = exec('sudo /usr/sbin/sendmail -bp -OQueueDirectory=/var/spool/mqueue.in 2>&1');
-                    //preg_match"/(Total requests: )(.*)/", $cmd, $output_array);
-                    //$inq = $output_array[2];
-                    //$cmd = exec('sudo /usr/sbin/sendmail -bp 2>&1');
-                    //preg_match"/(Total requests: )(.*)/", $cmd, $output_array);
-                    //$outq = $output_array[2];
-                    $inq = database::mysqli_result(dbquery('SELECT COUNT(*) FROM inq WHERE ' . $_SESSION['global_filter']),
-                        0);
-                    $outq = database::mysqli_result(dbquery('SELECT COUNT(*) FROM outq WHERE ' . $_SESSION['global_filter']),
-                        0);
+                    $pqerror .= 'XML-RPC Error: ' . $rsp->faultString();
                 }
-                echo '    <tr><td colspan="3" class="heading" align="center">' . __('mailqueue03') . '</td></tr>' . "\n";
-                echo '    <tr><td colspan="2"><a href="mailq.php?token=' . $_SESSION['token'] . '&amp;queue=inq">' . __('inbound03') . '</a></td><td align="right">' . $inq . '</td>' . "\n";
-                echo '    <tr><td colspan="2"><a href="mailq.php?token=' . $_SESSION['token'] . '&amp;queue=outq">' . __('outbound03') . '</a></td><td align="right">' . $outq . '</td>' . "\n";
             }
-
-            if (!DISTRIBUTED_SETUP) {
-                // Drive display
-                echo '    <tr><td colspan="3" class="heading" align="center">' . __('freedspace03') . '</td></tr>' . "\n";
-                foreach (get_disks() as $disk) {
-                    $free_space = disk_free_space($disk['mountpoint']);
-                    $total_space = disk_total_space($disk['mountpoint']);
-                    $percent = '<span>';
-                    if (round($free_space / $total_space, 2) <= 0.1) {
-                        $percent = '<span style="color:red">';
-                    }
-                    $percent .= ' [';
-                    $percent .= round($free_space / $total_space, 2) * 100;
-                    $percent .= '%] ';
-                    $percent .= '</span>';
-                    echo '    <tr><td>' . $disk['mountpoint'] . '</td><td colspan="2" align="right">' . formatSize($free_space) . $percent . '</td>' . "\n";
-                }
+            if ($pqerror !== '') {
+                echo '    <tr><td colspan="3">Warning: An error occured:' . $pqerror . '</td>' . "\n";
```

Comment:
We need to add a space after "occurred:".

`echo '    <tr><td colspan="3">Warning: An error occured:' .  ' ' . $pqerror . '</td>' . "\n";`